### DISCUSSION TITLE
ExploitPatcher book and shulker book ban

### DISCUSSION THREAD

**Discussion by Yukylyu** [Zrips/CMI#5609] [Feature Requests]
Hi could you block please the written_books and writable_book in shulkers to prevent client / server crashes?
book ban my video https://www.youtube.com/watch?v=vs5YbRthDq0
(i hope a blacklist items for shulkerbox)^^

example:
https://www.spigotmc.org/threads/book-ban.413093/
https://www.spigotmc.org/threads/banning-with-shulker-boxes.363241/?__cf_chl_jschl_tk__=40a175d17e5f9279adb549c2d51d8ae191da390a-1621549204-0-AVg0SqPyFLcamSvtOQi-CjDel_c5A0_jvnYiOgaSn4vREPt0csXFp6nrdBvrZ6VzONlj0oKzg_VOzlRKueloxbgb0hpFjt2RKARDUgc1u1a4T1MM84kzNFFJHObJ2y6LIHdwMUtNi6tXAg6q3ORBfVlvGPpHlfrETKOp3BpqvDa2cn7QfRIjXtAcCQfFEYFajdH0Ms_QF3Nr7t3AuXrR2X0_LMJJ1FVWqxVp2m8QghxrWgEgLPzJ93o1i9T__S01Q3eIDs-yz7z3ncRw8vpfOFgYzjxEoRLyMRjphxqdcYY2Nl-UX_Fzl49d8rfEarc3rSbFIoEJEuoLxFVCOQdCfGuAaEPYT0eniYTt37bvMc92s1Hq25Uekj_Rxk6uhmGjEdzPq-8CVo-G0kHRpxWFtJNWDmZMWNYGlkjPYk1IDADIzNgnQoNtUuQvbtgVgsksHaeC-ozgPPhGXIKrL43OEig

https://www.youtube.com/watch?v=r02ZqO9Vvoo
https://www.youtube.com/watch?v=JDGx8zeCsUA

I use worldguard for blacklist written_books and writable_book but it would be great if cmi could prevent this.

**Comment by CloudeLecaw** [comment]
your sure this needs an extra exploit?

if i would see a player doing this in creatinve move (its would take a bit endless long to do this in survival) i would direcly ban him. this has nothing to with testing as you can see it was their target here to crash their playeracc.
People like this should not play on your server ;)

**maybe the simplest solution would be adding a blacklist for items that get block into shulker boxes and done**
(maybe with a bypass for admins - i aspect no admin would do a crazzy thing like that ;) )

**Comment by Yukylyu** [comment]
My players have already done this in survival on my server, and yes it's possible on spigot 1.16.5 
look at my video => https://www.youtube.com/watch?v=vs5YbRthDq0
(sorry 0min25 => 0min40 the video has bug)

(If you have a plugin to blacklist certain items in a shulker, I would like it ^^ @CloudeLecaw )

**Comment by CloudeLecaw** [comment]
The exploit is called book ban exploit and is possible in all versions since shulkers are added to the game.

as a current solution we used this one
https://www.spigotmc.org/resources/dupe-fixes-illegal-stack-remover.44411/update?update=299041
(need also this https://www.spigotmc.org/resources/nbt-api.7939/)

i really hope it can be added to CMI soon as  a Blacklist for shulkeritems
